### PR TITLE
store caseReference in the camunda task

### PR DESCRIPTION
**Pull request by iguannago**
rename dueDate in the EventInformation.java to dateTime

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/resources/application-functional.yaml [hunk 1]
```diff
@@ -1,2 +1,10 @@
 targets:
   instance: ${TEST_URL:http://localhost:8088}
+  camunda: ${CAMUNDA_URL:http://camunda-bpm/engine-rest}
+
+idam:
+  s2s-auth:
+    url: ${S2S_URL:http://service-auth-provider-api}
+    secret: ${S2S_SECRET_WORKFLOW_API:AAAAAAAAAAAAAAAC}
+    name: ${S2S_NAME_WORKFLOW_API:wa_workflow_api}
```
Comment (alexbaeza) [inline src/functionalTest/resources/application-functional.yaml:9]: should this be the case-event handler

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -10,7 +10,8 @@ def secrets = [
     'wa-${env}' : [
     ],
     's2s-${env}': [
-        secret('microservicekey-wa-case-event-handler', 'S2S_SECRET_CASE_EVENT_HANDLER')
+        secret('microservicekey-wa-case-event-handler', 'S2S_SECRET_CASE_EVENT_HANDLER'),
+        secret('microservicekey-wa-workflow-api', 'S2S_SECRET_WORKFLOW_API'),
```
Comment (alexbaeza) [inline Jenkinsfile_CNP:14]: Do we need the workflow api secret

File: src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/InitiationTaskHandlerTest.java [hunk 3]
```diff
@@ -26,37 +30,44 @@
...
-            .jurisdictionId("IA")
-            .caseTypeId("Asylum")
-            .build();
+            requestParameters
+        )).thenReturn(new EvaluateDmnResponse<>(Collections.emptyList()));
 
         handlerService.evaluateDmn(eventInformation);
 
         Mockito.verify(apiClientToInitiateTask).evaluateDmn(
             eq("getTask_IA_Asylum"),
```
Comment (alexbaeza) [inline src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/InitiationTaskHandlerTest.java:66]: these should now be wa-task-initiation-ia-asylum, it's also case sensitive

Comment (iguannago) [reply src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/InitiationTaskHandlerTest.java]: done

File: Jenkinsfile_CNP (hunk 2 above)
Comment (iguannago) [reply Jenkinsfile_CNP:14]: TODO: raise PR with rpe

File: src/functionalTest/resources/application-functional.yaml (hunk 1 above)
Comment (iguannago) [reply src/functionalTest/resources/application-functional.yaml:9]: TODO: raise PR to rpe

File: Jenkinsfile_CNP (hunk 2 above)
Comment (iguannago) [reply Jenkinsfile_CNP:14]: https://github.com/hmcts/camunda-bpm/pull/84

File: src/functionalTest/resources/application-functional.yaml (hunk 1 above)
Comment (iguannago) [reply src/functionalTest/resources/application-functional.yaml:9]: https://github.com/hmcts/camunda-bpm/pull/84

File: src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java [hunk 4]
```diff
@@ -34,5 +40,35 @@ public void given_validEventInformation_should_respond_with_200() {
...
+    @After
+    public void cleanUpTask() {
+
+        given()
+            .contentType(APPLICATION_JSON_VALUE)
+            .accept(APPLICATION_JSON_VALUE)
+            .header(SERVICE_AUTHORIZATION, authTokenGenerator.generate())
+            .when()
+            .post(camundaUrl + "/task/{task-id}/complete", taskId);
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java:72]: Can we check the state of the completed task? I.e. check that the task no longer exists as part of completing the FT scenario?

File: src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java [hunk 5]
```diff
@@ -1,27 +1,33 @@
...
+import static org.hamcrest.CoreMatchers.is;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static uk.gov.hmcts.reform.wacaseeventhandler.CreatorObjectMapper.asJsonString;
 
 public class CaseEventHandlerControllerTest extends SpringBootFunctionalBaseTest {
 
+    private final String caseReference = UUID.randomUUID().toString();
+    private String taskId;
+
     @Test
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java:22]: Do we need a test to cover the invalidEventInformation scenario? I recall we discussed this but couldn't spot the test in FT on IT?

**Review by alexbaeza — APPROVED** [review]

**Review by alexbaeza — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndPointTest.java [hunk 6]
```diff
@@ -96,10 +97,10 @@ void given_message_then_return_expected_status_code(Scenario scenario) throws Ex
     private static Stream<Scenario> scenarioProvider() {
         EventInformation validEventInformation = EventInformation.builder()
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerEndPointTest.java:98]: Is this a duplicate? Functional test has a test for the happy path.

**Pull request merged**